woocommerce/woocommerce-ios
# Show specific message when opening a deleted product

## What
This PR adds a specific message to the user in case they try to open a product with invalid id, which most likely means that the product is deleted.

We already had a separate screen to fetch a product (`ProductLoaderViewController`) which shows a failure overlay with retry button in case fetching wasn't successful. 

In case we know that the id we're trying to fetch is invalid it doesn't make much sense to retry. In this case we're reusing existing failure overlay, but with a different message and without retry button.

Closes #2901 

## Test
1. Create an order
2. Delete one of the products that's in the order (also from trash)
3. Open the order in the app
4. Tap on the product deleted in 2.

- [ ]  Check that the error message is shown saying that the product is deleted

## Screenshots

|||
|-|-|
|![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 14 58 17](https://user-images.githubusercontent.com/54751/101498296-37c76e00-396c-11eb-8d8b-7b577e9b6e41.png)|![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 14 58 28](https://user-images.githubusercontent.com/54751/101498322-41e96c80-396c-11eb-892e-cb03fbc79803.png)|
|![Simulator Screen Shot - iPhone 11 - 2020-12-08 at 14 58 34](https://user-images.githubusercontent.com/54751/101498327-431a9980-396c-11eb-942b-a4b825599b9a.png)|![Simulator Screen Shot - iPad Pro (9 7-inch) - 2020-12-08 at 15 01 41](https://user-images.githubusercontent.com/54751/101498331-43b33000-396c-11eb-93a4-5b43da16fa8b.png)|

Update release notes:
